### ISSUE TITLE
Thunk action added to the queue just once

### ISSUE THREAD

**Issue by jura1514** [rgommezz/react-native-offline#163]
Hi,

Have a question regarding how actions are added to the queue.
Sorry for asking it here, but I thought I should get quicker reply here then on Stackoverflow

Basically, when my app is offline and I click on a button to trigger a thunk action to send a request to create/update a record, this function is added to the queue just once.

Is it a normal behavior? Cause in the example app I was able to add same actions to the queue.

Thanks in advance.

**Comment by rgommezz** [comment]
Hi @jura1514, 

That's the normal behaviour. In the example the actions `ADD_1` and `SUB_1` are added multiple times because the [action objects are different](https://github.com/rgommezz/react-native-offline/blob/master/example/redux/actions.js#L5) every time a new one is dispatched.
 
I just wanted to illustrate the different behaviours. If you long press on an action, you should see a dialog that explains it in details.

Hope that clarifies it.

**Comment by jura1514** [comment]
Hi,

Thanks for your replay.

Is there any work around for being able to add same thunk action to the queue or is it something perhaps might be added in a future?

**Comment by rahulnair25283** [comment]
For lack of a better place to ask this question, @jura1514 did you manage to find a way to have multiple thunks of the same type e.g. ADD_1 to the offline queue?

**Comment by nicemanis** [comment]
I also need to be able to add multiple thunks. It will overwrite the previous thunk even if the arguments are different.
@rgommezz is it possible to configure if the action/thunk should be overwritten or not in the queue?

**Comment by jura1514** [comment]
@rahulnair25283 Unfortunatelly I didn't, I decided to use different approach, without using this library.

**Comment by nicemanis** [comment]
@rgommezz Would it be possible to introduce the possibility of adding multiple thunks?

**Comment by gabsima** [comment]
There seems to be a problem in the reducer's method : handleOfflineAction. If there is a similar action in queue, it is removed. Then the new action is added. 

I'll make a PR to configure this behavior.

**Comment by gabsima-nexapp** [comment]
It's actually `getSimilarActionInQueue`. With thunks, even if the payload is different for each function, the signature `.toString()` is the same because the payload is given from a closure to the action.

This behavior is making every action in thunk with the same type to be replaced with the newest one. Even if the payloads are different. I dont't have any idea for a solution to this problem. I tried accessing the context of the functions to compare them, but i wasn't able. I also tried to modify the actions to have the payload as parameters but it doesn't work either.
